File: src/main/java/com/gitmanager/model/BranchInfo.java
package com.gitmanager.model;

public class BranchInfo {

    private final String name;
    private final String commitId;
    private final boolean remote;
    private final boolean head;

    public BranchInfo(String name, String commitId, boolean remote, boolean head) {
        this.name = name;
        this.commitId = commitId;
        this.remote = remote;
        this.head = head;
    }

    public String getName() { return name; }
    public String getCommitId() { return commitId; }
    public boolean isRemote() { return remote; }
    public boolean isHead() { return head; }

    @Override
    public String toString() {
        return name + (remote ? " (remoto)" : "");
    }
}

File: frontend/lib/models/branch_info.dart
class BranchInfo {
  final String name;
  final String commitId;
  final bool remote;
  final bool head;

  BranchInfo({
    required this.name,
    required this.commitId,
    this.remote = false,
    this.head = false,
  });

  factory BranchInfo.fromJson(Map<String, dynamic> json) {
    return BranchInfo(
      name: json['name'] ?? '',
      commitId: json['commitId'] ?? '',
      remote: json['remote'] ?? false,
      head: json['head'] ?? false,
    );
  }
}
